src/db-api.js

```diff
@@ -68,47 +68,141 @@ const transactionsHistoryForAddresses = (db: Pool) => async (
 
 // The remaining queries should be used only for the purposes of the legacy API!
 
-/**
- * Queries DB looking for successful transactions associated with any of the given addresses.
- * @param {Db Object} db
- * @param {Array<Address>} addresses
- */
-const bulkAddressSummary = (db: Pool) => async (addresses: Array<string>): Promise<ResultSet> =>
-  db.query({
-    text: `SELECT * FROM "txs"
-      WHERE hash = ANY (SELECT tx_hash FROM "tx_addresses" WHERE address = ANY($1))
-      AND tx_state = $2
-      ORDER BY time DESC`,
-    values: [addresses, 'Successful'],
-  })
-
 /**
 * Queries TXS table looking for a successful transaction with a given hash
 * @param {Db Object} db
-* @param {*} tx
+* @param {Transaction} tx
 */
 const txSummary = (db: Pool) => async (tx: string): Promise<ResultSet> =>
   db.query({
     text: 'SELECT * FROM "txs" WHERE hash = $1 AND tx_state = $2',
     values: [tx, 'Successful'],
   })
 
+/**
+* Queries TX table looking for tx by its hash
+* @param {Db Object} db
+* @param {Transaction} tx
+*/
+const getTx = (db: Pool) => async (tx: string): Promise<ResultSet> =>
+  db.query({
```

Comment:
are you sure you don't need to filter out pending/failed transactions as it was done in the previous version of this call?

Reply:
There is no mention about tx success in the database
![image](https://user-images.githubusercontent.com/22454337/85681685-d10b0c80-b6cb-11ea-8c51-c134fb3465c7.png)